nvflare/fuel/utils/class_utils.py

```diff
@@ -96,13 +111,31 @@ def _create_classes_table(self):
                                         and inspect.isclass(obj)
                                         and obj.__module__ == module_name
                                     ):
-                                        self._class_table[name] = module_name
+                                        scan_result = self._class_table.get(name, _ModuleScanResult(class_name=name))
+                                        scan_result.add_module(module_name)
+                                        self._class_table[name] = scan_result
                             except (ModuleNotFoundError, RuntimeError) as e:
-                                self._logger.warning(
+                                self._logger.debug(
                                     f"Try to import module {module_name}, but failed: {e}. "
-                                    f"Please ignore this if you are not using files in module: {module_name}."
+                                    f"Can't use name in config to refer to classes in module: {module_name}."
                                 )
                                 pass
 
-    def get_module_name(self, class_name):
-        return self._class_table.get(class_name, None)
+    def get_module_name(self, class_name) -> Optional[str]:
+        """Gets the name of the module that contains this class.
+
+        Args:
+            class_name: The name of the class
+
+        Returns:
+            The module name if found.
+        """
+        if class_name not in self._class_table:
+            return None
+        scan_result = self._class_table[class_name]
+        module_name = scan_result.modules[0]
+        if len(scan_result.modules) != 1:
+            self._logger.warning(
+                f"More than 1 module contains class {class_name}. The first module: {module_name} is used."
+            )
```

Comment:
The intention of this is to address the duplicate class name used in different module. But the solution is half baked. Simply giving the warning for this could easily cause confusion, because a wrong class instance been created. 

Instead, it's better to let the configuration fail and let the user know the situation. At the same time, we can provide the solution for choose which one to use. For example, when duplicate class name detected, we store the "module_name.class_name" as the key in the _class_table. So user can use that unique name in the configuration.